File: 1-two-sum/two-sum.cs
// public class Solution {
//     public int[] TwoSum(int[] nums, int target) {
//         for(int i=0;i<nums.Length;i++){
//             for(int j=i+1;j<nums.Length;j++){
//                 if(nums[i]+nums[j] == target){
//                     return new int[] {i,j};
//                 }
//             }
//         }
//         return new int[]{};
//     }
// }

/////////////////


public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int , int> dict = new Dictionary<int ,int>();
        for(int i=0;i<nums.Length;i++){
            int rem  = target - nums[i];
            if(dict.ContainsKey(rem)){
                return new int[] {dict[rem] , i};
            }
            dict[nums[i]] = i;
        }
        return new int[] {};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        n = len(nums)
        i = 0
        while(i < n-1):
            j = i+1
            while(j < n):
                if(nums[i] + nums[j] == target):
                    ans.append(i)
                    ans.append(j)
                    return ans
                j += 1

            i += 1

        return ans
        


            

